Integrations/src/main/java/io/deephaven/integrations/learn/gather/NumPy.java

```diff
@@ -0,0 +1,668 @@
+package io.deephaven.integrations.learn.gather;
+
+import io.deephaven.chunk.*;
+import io.deephaven.chunk.attributes.Values;
+import io.deephaven.engine.rowset.RowSequence;
+import io.deephaven.engine.table.ChunkSource;
+import io.deephaven.engine.table.ColumnSource;
+import io.deephaven.engine.table.SharedContext;
+import io.deephaven.util.SafeCloseableList;
+
+/**
+ * Gatherer takes Deephaven columnar data and places it into a buffer to be used by Python. The Python object will take
+ * data from the buffer and use it to construct a 2d array of specified size.
+ */
+public class NumPy {
+
+    private static final int COPY_CHUNK_SIZE = 2048;
+
+    /**
+     * Copy data from a table into a 2d tensor of Booleans.
+     *
+     * @param rowSeq indices of the rows of the table to put into the tensor
+     * @param columnSources columns of data to put into the tensor
+     * @param ColumnMajorOrder the major order of the operation
+     * @return contiguous RAM allocated for the tensor
+     */
+    public static boolean[] tensorBuffer2DBoolean(final RowSequence rowSeq, final ColumnSource<?>[] columnSources,
+            boolean ColumnMajorOrder) {
+        if (ColumnMajorOrder) {
+            boolean[] tensor = tensorBuffer2DBooleanColumnMajor(rowSeq, columnSources);
+            return tensor;
+        } else {
+            boolean[] tensor = tensorBuffer2DBooleanRowMajor(rowSeq, columnSources);
+            return tensor;
+        }
+    }
+
+    /**
+     * Copy data from a table into a 2d tensor of Booleans in column-major order.
+     *
+     * @param rowSeq indices of the rows of the table to put into the tensor
+     * @param columnSources columns of data to put into the tensor
+     * @return contiguous RAM allocated for the tensor
+     */
+    public static boolean[] tensorBuffer2DBooleanColumnMajor(final RowSequence rowSeq,
```

Comment:
Should the `ColumnMajor` and `RowMajor` functions be public or private?  I don't have a strong view, but have a slight lean to minimizing the exposed API.